src/codegen/templates/_grpc_interpreter.py.mako

```diff
@@ -143,6 +143,14 @@ class GrpcStubInterpreter(BaseInterpreter):
         else:
             self._raise_error(error_code, error_message, samps_per_chan_written, samps_per_chan_read)
 
+    def _check_for_error_from_response(self, error_code, samps_per_chan_written=None, samps_per_chan_read=None):
+        if error_code != 0:
+            if error_code in ERROR_MESSAGES:
+                error_message = ERROR_MESSAGES.get(error_code)
+            else:
+                error_message = self.get_error_string(error_code)
```

Comment:
Also add a comment that this is an optimization for partial reads, so that someone doesn't refactor it into `error_message = ERROR_MESSAGES.get(error_code, self.get_error_string(error_code))`